### PR TITLE
Agreement: Support payable actions

### PR DESCRIPTION
**Pull request by facuspagnuolo**
This PR basically implements the new forwarding interface (payable + context) in the disputable app used to test the agreement app, forcing the later to handle ETH fees.

### CODE REVIEW

File: apps/agreement/contracts/example/RegistryApp.sol [hunk 1]
```diff
@@ -121,23 +122,29 @@ contract Registry is DisputableAragonApp {
...
     * @return True if the given address can submit actions, false otherwise
     */
-    function canForward(address _sender, bytes _data) public view returns (bool) {
-        (bytes32 id,) = _decodeData(_data);
-        return canPerform(_sender, REGISTER_ENTRY_ROLE, arr(id));
+    function canForward(address _sender, bytes _data) external view returns (bool) {
+        return _canForward(_sender, _data);
+    }
+
+    // TODO: how are we going to implement this from the disputable apps?
```
Comment (facuspagnuolo) [inline apps/agreement/contracts/example/RegistryApp.sol:145]: This is an important thing that came up we will need to figure out soon

File: apps/agreement/contracts/Agreement.sol [hunk 2]
```diff
@@ -773,16 +775,23 @@ contract Agreement is IAgreement, AragonApp {
             return;
         }
 
-        // Get staking pool
-        Staking staking = stakingFactory.getOrCreateInstance(token);
-
-        // Pull required fee amount from staking pool
-        _lockBalance(staking, _submitter, amount);
-        _slashBalance(staking, _submitter, address(this), amount);
+        uint256 sentEth = 0;
```
Comment (bingen) [inline apps/agreement/contracts/Agreement.sol:778]: Why do we need this?

File: apps/agreement/contracts/example/RegistryApp.sol [hunk 3]
```diff
@@ -121,23 +122,29 @@ contract Registry is DisputableAragonApp {
     * @notice Schedule a new entry
     * @dev IForwarder interface conformance
     * @param _data Data requested to be registered
+    * @param _context Link to a human-readable text giving context for the given action
     */
-    function forward(bytes memory _data) public {
```
Comment (bingen) [inline apps/agreement/contracts/example/RegistryApp.sol:125]: Why are we removing the `memory` keyword?

File: apps/agreement/contracts/example/RegistryApp.sol [hunk 4]
```diff
@@ -121,23 +122,29 @@ contract Registry is DisputableAragonApp {
...
-        require(canForward(msg.sender, _data), ERROR_CANNOT_REGISTER);
+    function forward(bytes _data, bytes _context) external payable {
```
Comment (bingen) [inline apps/agreement/contracts/example/RegistryApp.sol:127]: Isn’t first parameter an EVM script here? That’s confusing to me. Why do we need forwarding then?

File: apps/agreement/contracts/example/RegistryApp.sol (hunk 3 above)
Comment (facuspagnuolo) [reply apps/agreement/contracts/example/RegistryApp.sol:125]: It is calldata now that it is external :)

File: apps/agreement/contracts/Agreement.sol (hunk 2 above)
Comment (facuspagnuolo) [reply apps/agreement/contracts/Agreement.sol:778]: Simply to support both ETH or ERC20 fees

File: apps/agreement/contracts/example/RegistryApp.sol (hunk 4 above)
Comment (facuspagnuolo) [reply apps/agreement/contracts/example/RegistryApp.sol:127]: Yes it is, but I'm playing with it in the Registry app. Although it is a sample app, we can actually not make it forwardable

File: apps/agreement/contracts/example/RegistryApp.sol (hunk 3 above)
Comment (bingen) [reply apps/agreement/contracts/example/RegistryApp.sol:125]: Oh, of course, haha! :facepalm:

File: apps/agreement/contracts/Agreement.sol (hunk 2 above)
Comment (bingen) [reply apps/agreement/contracts/Agreement.sol:778]: But can’t you use `msg.value` everywhere?

Comment (facuspagnuolo) [reply apps/agreement/contracts/Agreement.sol:778]: True 😅

**Pull request merged**